### DISCUSSION TITLE
word order/grammar rules

### DISCUSSION THREAD

**Discussion by JoaoVictor834** [dfint/.github#5] [Q&A] [answered]
Hi! first sorry for the bad english, i'm using google translator

Let's get to the point: Many languages ​​have different grammar rules, like word order, I wanted to know if there is any way to change at least the word order, like in my language:

English:
wooden bin

adj + noun

Portuguese:
wooden bin
caixa de madeira

noun + preposition + adj

In translation it is:
caixa madeira

adj + noun

Would there be any way to change this?

**Comment by insolor** [comment, marked as answer]
For now, no, this is not possible. For previous versions of DF (before 50) we used interception of text right before rendering in SDL_ttf.dll to replace it with a modified text (e.g. changing word order, changing forms of words). Now the game doesn't not use SDL_ttf.dll, so this is not easy anymore.

We have some results on bringing SDL_ttf back (this is also important to support languages with hieroglyphic writing systems, see discussions in the parallel thread: https://github.com/orgs/dfint/discussions/1), but I cannot promise that this will be ready to use soon. But this is on our plans.

**Comment by JoaoVictor834** [comment]
Thanks for the answer! Sad that this method no longer works, now I hope you can get around it, congratulations on the project, it's helping a lot of people!
